src/ast/passes/pid_filter_pass.cpp

```diff
@@ -85,8 +87,7 @@ void PidFilterPass::visit(Probe &probe)
 
   for (AttachPoint *ap : probe.attach_points) {
     if (probe_needs_pid_filter(ap)) {
-      probe.block->stmts.insert(probe.block->stmts.begin(),
-                                create_pid_filter(ast_, *pid, probe.loc));
+      probe.block = create_pid_filter(ast_, *pid, probe.block);
```

Comment:
Was this needed? An inserted `IfExpr` with a `return` in the `then` should have worked right?

Reply:
kk - NBD, was just curious.